python/setup.py

```diff
@@ -37,6 +37,7 @@ def get_lib_path():
   packages = find_packages(),
   install_requires=[
       'numpy==1.18.5',
+      'scipy==1.7.2',
```

Comment:
Where scipy is used?

Reply:
This was added to resolve a CI test issue: scipy is a secondary dependency, and sometimes the python package manager automatically selects scipy==1.8.3, which needs python3.8 to run, and causes an error in the CI.

Reply:
The scipy dependency was moved to hlib by another PR in the main branch, I have merged with changes and the dependency issue is resolved